### PR TITLE
[ACR] Support disable public network access

### PR DESCRIPTION
**Pull request by yugangw-msft**
This is pending SDK release

**Description**  
Support  `--public-network-access-enabled`
Improve tab output of `az acr show-endpoints`

**Testing Guide**  

az acr update/create --public-network-access-enabled

CC @djyou 

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/acr/_params.py [hunk 1]
```diff
@@ -79,12 +79,13 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
         with self.argument_context(scope, arg_group='Network Rule') as c:
             c.argument('default_action', arg_type=get_enum_type(DefaultAction),
                        help='Default action to apply when no rule matches. Only applicable to Premium SKU.')
+            c.argument('public_network_access_enabled', get_three_state_flag(), help="Allow public network access for the container registry. The Default is allowed")
```
Comment (djyou) [inline src/azure-cli/azure/cli/command_modules/acr/_params.py:82]: I saw a few other services used `--enable-public-network`, should we be consistent?

https://github.com/Azure/azure-cli/blob/3315616902b390805ca8d138a6fa168ff0518caf/src/azure-cli/azure/cli/command_modules/cosmosdb/_params.py#L67

https://github.com/Azure/azure-cli/blob/015be59dfb57fbaf9d5215d49fe0488ea41d3a53/src/azure-cli/azure/cli/command_modules/sql/_params.py#L1134

File: src/azure-cli/azure/cli/command_modules/acr/_docker_utils.py [hunk 2]
```diff
@@ -332,6 +332,7 @@ def _get_credentials(cmd,  # pylint: disable=too-many-statements
         if challenge.status_code == 403:
             raise CLIError("Looks like you don't have access to registry '{}'. "
                            "To see configured firewall rules, run 'az acr show --query networkRuleSet --name {}'. "
+                           "To see public network access, run 'az acr show --query publicNetworkAccess'."
```
Comment (djyou) [inline src/azure-cli/azure/cli/command_modules/acr/_docker_utils.py:335]: nit:
To see public network access status ...
To see if public network access is enabled ...

File: src/azure-cli/azure/cli/command_modules/acr/_params.py (hunk 1 above)
Comment (yugangw-msft) [reply src/azure-cli/azure/cli/command_modules/acr/_params.py:82]: Agreed, let us be consistent

**Comment by yonzhan** [comment]
acr

File: src/azure-cli/azure/cli/command_modules/acr/_params.py (hunk 1 above)
Comment (yugangw-msft) [reply src/azure-cli/azure/cli/command_modules/acr/_params.py]: Per conversation with Mario from NRP, we can use `--public-network-enabled` for consistency with other ACR arguments

**Comment by yugangw-msft** [comment]
@fengzhou-msft, review comments were addressed. If you are happy, can you please approve and merge it?

**Pull request merged**